coalib/parsing/DefaultArgParser.py

```diff
@@ -110,6 +110,21 @@ def default_arg_parser(formatter_class=None):
                               metavar='FILE',
                               help=SAVE_HELP)
 
+    CACHING_HELP = ('Run coala only on files that have changed '
+                    'since the last time coala was run. Note: '
+                    'Caching is currently experimental and '
+                    'will be enabled by default from the next '
+                    'release (this option will be removed)')
+    config_group.add_argument('-C',
+                              '--caching',
+                              const=True,
+                              action='store_const',
+                              help=CACHING_HELP)
+    config_group.add_argument('--flush-cache',
+                              const=True,
+                              action='store_const',
+                              help='Rebuild the file cache')
```

Comment:
I'm in favor of having it under Inputs as well. I wonder though if the name could be a bit more descriptive. The word 'caching' is pretty broad and does not make it apparent that it won't rerun unchanged files. Maybe something like `--changed-only` or `--incremental`.